File: com.systemincloud.examples.tasks.javatask/src/test/java/tasks/ports/Atomic3SIn0AIn2Out.java
package tasks.ports;

import com.systemincloud.modeler.tasks.javatask.api.InputPort;
import com.systemincloud.modeler.tasks.javatask.api.JavaTask;
import com.systemincloud.modeler.tasks.javatask.api.OutputPort;
import com.systemincloud.modeler.tasks.javatask.api.annotations.InputPortInfo;
import com.systemincloud.modeler.tasks.javatask.api.annotations.JavaTaskInfo;
import com.systemincloud.modeler.tasks.javatask.api.annotations.OutputPortInfo;
import com.systemincloud.modeler.tasks.javatask.api.data.Float32;
import com.systemincloud.modeler.tasks.javatask.api.data.Float64;
import com.systemincloud.modeler.tasks.javatask.api.data.Int32;
import com.systemincloud.modeler.tasks.javatask.api.data.Text;

@JavaTaskInfo
public class Atomic3SIn0AIn2Out extends JavaTask {

	@InputPortInfo(name = "In1", dataType = Float64.class)
	public InputPort in1;
	@InputPortInfo(name = "In2", dataType = Float32.class)
	public InputPort in2;
	@InputPortInfo(name = "In3", dataType = Int32.class)
	public InputPort in3;
	@OutputPortInfo(name = "Out1", dataType = Text.class)
	public OutputPort out1;
	@OutputPortInfo(name = "Out2", dataType = Int32.class)
	public OutputPort out2;

	@Override
	public void execute(int grp) {
		Float64 in1Text = in1.getData(Float64.class);
		Float32 in2Text = in2.getData(Float32.class);
		Int32 in3Text = in3.getData(Int32.class);
		double in1Double = in1Text.getValue();
		float  in2Float = in2Text.getValue();
		double sum = in1Double + (double) in2Float;
		out1.putData(new Text(in1Double + " + " + in2Float + " = " + sum));
		out2.putData(new Int32(in3Text.getValue()));
	}

}

File: com.systemincloud.examples.tasks.pythontask/src/test/py/tasks/ports/Atomic3SIn0AIn2Out.py
from sicpythontask.PythonTaskInfo import PythonTaskInfo
from sicpythontask.PythonTask import PythonTask
from sicpythontask.InputPort import InputPort
from sicpythontask.OutputPort import OutputPort
from sicpythontask.data.Int32 import Int32
from sicpythontask.data.Float32 import Float32
from sicpythontask.data.Float64 import Float64
from sicpythontask.data.Text import Text

@PythonTaskInfo
class Atomic3SIn0AIn2Out(PythonTask):

    def __init_ports__(self):
        self.in1 = InputPort(name="in1", data_type=Float64)
        self.in2 = InputPort(name="in2", data_type=Float32)
        self.in3 = InputPort(name="in3", data_type=Int32)
        self.out1 = OutputPort(name="out1", data_type=Int32)
        self.out2 = OutputPort(name="out2", data_type=Text)

    def execute(self, grp):
        in1Text = self.in1.get_data(Float64)
        in2Text = self.in2.get_data(Float32)
        in3Text = self.in3.get_data(Int32)
        in1Double = in1Text.value
        in2Float = in2Text.value
        sum_ = in1Double + in2Float
        self.out2.put_data(Text(str(in1Double) + " + " + str(in2Float) + " = " + str(sum_)))
        self.out1.put_data(Int32(in3Text.value));

